### PR TITLE
docs: Improve user documentation for supported operators and expressions

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

The list of supported expressions was incomplete and we did provide any information about known incompatibilities

## What changes are included in this PR?

- Use table instead of list
- Add more expressions
- Add notes about know incompatibilities
- Renamed `Negative` to `UnaryMinus` in protobuf for consistency

### CODE REVIEW

**Comment by andygrove** [comment]
I plan on keeping this as draft for a while and will keep updating it as we prepare for the 0.1.0 release.

File: core/src/execution/datafusion/planner.rs
Comment (andygrove) [inline core/src/execution/datafusion/planner.rs:566]: I renamed this because it was inconsistent with Spark naming

**Comment by andygrove** [comment]
@parthchandra @kazuyukitanimura @huaxingao @viirya This is ready for review now. Thanks.

**Comment by andygrove** [comment]
@comphead This is the user guide version of the list of supported expressions with notes about known issues. I think this is separate from the list that you are generating for the contributors guide.

File: docs/source/user-guide/operators.md
Comment (comphead) [inline docs/source/user-guide/operators.md:33]: oh, is SMJ disabled by default?

Comment (andygrove) [reply docs/source/user-guide/operators.md]: It looks like I misunderstood this. I have update this.

File: docs/source/user-guide/operators.md
Comment (viirya) [inline docs/source/user-guide/operators.md:34]:
```suggestion
| Hash Join                                    |       |
| BroadcastHashJoinExec                        |       |
```

File: docs/source/user-guide/operators.md
Comment (viirya) [inline docs/source/user-guide/operators.md:36]:
```suggestion
| Expand                                       |       |
| Union                                        |       |
```

File: docs/source/user-guide/expressions.md
Comment (kazuyukitanimura) [inline docs/source/user-guide/expressions.md:86]: nit: `RLike` is still under review. Not sure which gets merged first...

**Review by kazuyukitanimura — APPROVED** [review]
LGTM pending ci

File: docs/source/user-guide/expressions.md
Comment (andygrove) [reply docs/source/user-guide/expressions.md]: I have removed this and will add it back in the RLike PR after this one is merged

**Pull request merged**
